.dd4hep-ci.d/compile_and_test.sh

```diff
@@ -13,8 +13,15 @@ cmake -GNinja -DDD4HEP_USE_GEANT4=ON \
     -DDD4HEP_DEBUG_CMAKE=ON \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_CXX_FLAGS="-fdiagnostics-color=always -Werror"  \
-    -DCMAKE_CXX_STANDARD=${STANDARD} ${CMAKE_ARGS} .. && \
-ninja && \
+    -DBUILD_SHARED_LIBS=${SHARED} \
+    -DCMAKE_CXX_STANDARD=${STANDARD} .. && \
+    ninja
+
+if [[ "${SHARED}" == "OFF" ]];
+then
+    exit 0;
```

Comment:
Like this it will always pass since there is no and and the last return code will be given to travis, I think maybe something like this would be better
```suggestion
    exit $?;
```

Reply:
Sorry, I was premature, the evaluation of the if statement overrides `$?` so you have to save the return value from the build command in a temp variable...

Reply:
Done, I think.

Reply:
In principle, I think you still have to `and` the build result with the rest of the CI chain after the if
```
BUILD_RESULT && \
ninja install && \ 
. ../bin/thisdd4hep.sh && \ 
...
```
otherwise the non static chains will not work properly. But it's probably not possible that `BUILD_RESULT` would be 1 and the subsequent CI commands would results in a positive result. So it's more a question of taste.

Reply:
Since `ninja install` will rerun compilation or linking, it might be that things are working in the second pass of the compilation attempt, which is not what we want. So I added a check for BUILD_RESULT